src/sources/futures.rs

```diff
@@ -56,43 +116,187 @@ impl<T> Scheduler<T> {
+        };
+
+        assert_send_and_sync(&schedule);
+
+        // Spawn the future.
+        let (runnable, task) = {
+            let builder = Builder::new().metadata(index);
+
+            // SAFETY: spawn_unchecked has four safety requirements:
+            //
+            // - "If future is not Send, its Runnable must be used and dropped on the original thread."
+            //
+            //   The runnable is created on the origin thread and sent to the origin thread, since both
+            //   Scheduler and Executor are !Send and !Sync. The waker may be sent to another thread,
+            //   which means that the scheduler function (and the Runnable it handles) can exist on
+            //   another thread. However, the scheduler function immediately sends it back to the origin
+            //   thread.
+            //
+            //   The issue then becomes "is the Runnable dropped unsoundly when the channel is closed?"
+            //   This problem is circumvented by immediately waking all runnables (pushes them into the
+            //   queue) and then draining the queue. This means that, before the channel is closed, all
+            //   runnables will be destroyed, preventing any unsoundness.
+            //
+            // - "If future is not 'static, borrowed variables must outlive its Runnable."
+            //
+            //   `F` is `'static`, so we don't have to worry about this one.
+            //
+            // - "If schedule is not Send and Sync, the task’s Waker must be used and dropped on the
+            //    original thread."
+            //
+            //   The schedule function uses a thread-safe MPSC channel to send the runnable back to the
+            //   origin thread. This means that the waker can be sent to another thread, which satisfies
+            //   this requirement.
+            //
+            // - "If schedule is not 'static, borrowed variables must outlive the task’s Waker."
+            //
+            //   `schedule` and the types it handles are `'static`, so we don't have to worry about
+            //   this one.
+            unsafe { builder.spawn_unchecked(move |_| future, schedule) }
+        };
+
+        // Insert the runnable into the set of active tasks.
+        active.insert(Active::Future(runnable.waker()));
+        drop(active_guard);
+
+        // Schedule the runnable and detach the task so it isn't cancellable.
+        runnable.schedule();
+        task.detach();
+
+        Ok(())
     }
 }
 
-/// Error generated when trying to schedule a future after the
-/// executor was destroyed.
-#[derive(thiserror::Error, Debug)]
-#[error("the executor was destroyed")]
-pub struct ExecutorDestroyed;
+impl Sender {
+    /// Send a runnable to the executor.
+    fn send(&self, runnable: Runnable<usize>) {
+        // Send on the channel.
+        if let Err(e) = self
+            .sender
+            .lock()
+            .unwrap_or_else(|e| e.into_inner())
```

Comment:
Would be better to have a comment explaining why ignoring the poisoning error is fine, if it is. And if not, to handle it properly.